src/main/java/gift/entity/Option.java

```diff
@@ -0,0 +1,70 @@
+package gift.entity;
+
+import gift.exception.OutOfQuantityException;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import jakarta.persistence.UniqueConstraint;
+
+@Entity
+@Table(name = "options", uniqueConstraints = {
+        @UniqueConstraint(
+                name = "UniqueOptionName",
+                columnNames = {"name", "product_id"}
+        )
+})
+public class Option {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+
+    private int quantity;
+
+    @ManyToOne(fetch = FetchType.LAZY) // 성능 향상 고려를 위해 지연 로딩 적용
+    @JoinColumn(name = "product_id", nullable = false)
+    private Product product;
+
+    protected Option() {}
+
+    public Option(String name, int quantity, Product product) {
+        this.name = name;
+        this.quantity = quantity;
+        this.product = product;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public Product getProduct() {
+        return product;
+    }
+
+    public void updateOption(String name, int quantity) {
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public void subtractQuantity(int amount) {
+        int remainingQuantity = this.quantity - amount;
+        if (remainingQuantity < 0) {
+            throw new OutOfQuantityException("수량이 부족합니다. (남은 수량: " + this.quantity + ")");
+        }
```

Comment:
개인적인 생각으로는 이게 조금 더 자연스럽지 않을까요? 

> 남은 재고가 차감 지정 숫자보다 적으면 예외를 던진다. 

```suggestion
        if (this.quantity < amount) {
            throw new OutOfQuantityException("수량이 부족합니다. (남은 수량: " + this.quantity + ")");
        }
        
        this.quantity - amount;
        
```

Reply:
제안 주신 것처럼 검증 후 값 수정하는 방식으로 수정하였습니다..!